cloudconfig/machinecloudconfig.go

```diff
@@ -0,0 +1,291 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package cloudconfig
+
+import (
+	"encoding/base64"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/juju/errors"
+	"github.com/juju/loggo"
+	"github.com/juju/utils"
+	utilsos "github.com/juju/utils/os"
+	utilsseries "github.com/juju/utils/series"
+	"gopkg.in/yaml.v2"
+
+	"github.com/juju/juju/juju/paths"
+)
+
+// GetMachineCloudInitData returns a map of all cloud init data on the machine.
+func GetMachineCloudInitData(series string) (map[string]interface{}, error) {
```

Comment:
Could we expand the comment to explain how series is used. A reader might wonder why series needs to be specified when it can be determined from the host machine. Maybe a TODO to explain how series will be used when we need to support trusty on xenial etc

Reply:
done